client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -72,6 +73,21 @@ export const requestJetpackOnboardingSettings = ( { dispatch, getState }, action
 	);
 };
 
+export const announceRequestFailure = ( { dispatch, getState }, { siteId } ) => {
+	const url = getUnconnectedSiteUrl( getState(), siteId );
+	if ( ! url ) {
+		return noop;
+	}
```

Comment:
Instead of doing nothing, you could still show the error notice, just without the link. Better UX IMO.

Reply:
Great idea, done in https://github.com/Automattic/wp-calypso/pull/22443/commits/32ee4dc9ef0eb7959a4ff7d907857ec4f359672d.